src/main/java/fizzbuzz/FizzBuzz.java

```diff
@@ -3,16 +3,24 @@
 
 package fizzbuzz;
 
+import java.util.List;
+
 public class FizzBuzz {
 
+    private List<FizzBuzzRule> rules = List.of(new FooRule(), //

```

Comment:
Creating one class per rule is nice. It seems it will be easy to extend the way rules work by changing this list.
I assume creating a ruleProvider class or something would work even better, so you do not have to touch this class when the rules change.